### ISSUE TITLE
babel.dates.format_timedelta throws KeyError exception on certain locale

### ISSUE THREAD

**Issue by khanguyen-yousician** [python-babel/babel#818]
## Overview Description
`babel.dates.format_timedelta` throws `KeyError` exception on certain `locale`. Version 2.7.0
## Steps to Reproduce

```
from datetime import timedelta
import babel
from babel.core import Locale
from babel.dates import format_timedelta


def try_format(locale):
    try:
        format_timedelta(
            timedelta(hours=1),
            granularity="hour",
            threshold=float("inf"),
            format="narrow",
            locale=locale,
        )
        return True
    except Exception as e:
        print('%s => %s: %s' % (locale, e.__class__.__name__, e))
    return False


from babel.localedata import locale_identifiers

failed = 0
for locale in sorted(locale_identifiers()):
    if not try_format(locale):
        failed += 1
```
## Actual Results
```
hr => KeyError: 'one'
hr_BA => KeyError: 'one'
hr_HR => KeyError: 'one'
pt => KeyError: 'one'
pt_BR => KeyError: 'one'
```

## Expected Results
No `KeyError` exception.

**Comment by akx** [comment]
Hi! The KeyError problem was fixed by #827, but not all of those locales have `narrow` data, so `format_timedelta()` would return an empty string.

**Comment by akx** [comment]
With #827 merged (i.e. with Babel 3.10), the repro script doesn't raise exceptions. Thank you for the report!
